static/app/views/issueList/overview.tsx

```diff
@@ -757,37 +759,72 @@ class IssueListOverview extends React.Component<Props, State> {
 
   onGroupChange() {
     const {organization} = this.props;
+    const {actionTakenIds} = this.state;
     const query = this.getQuery();
     const hasIssueListRemovalAction = organization.features.includes(
       'issue-list-removal-action'
     );
 
     // TODO(Kelly): update once issue-list-removal-action feature is stable
-    if (hasIssueListRemovalAction && !this.state.realtimeActive) {
-      const resolvedIds = this._streamManager
-        .getAllItems()
-        .filter(id => id.status === 'resolved')
-        .map(item => item.id);
-      const ignoredIds = this._streamManager
-        .getAllItems()
-        .filter(id => id.status === 'ignored')
-        .map(item => item.id);
-      const reviewedIds = this._streamManager
-        .getAllItems()
-        .filter(id => !id.inbox && id.status !== 'resolved' && id.status !== 'ignored')
-        .map(item => item.id);
-      // Remove Ignored and Resolved group ids from the issue stream, but if you have a query
-      // that includes these statuses or there's no query/you want to see ALL issues,
-      // don't trigger these group ids to be removed from the issue stream.
-      if (resolvedIds.length > 0 && !query.includes('is:resolved') && !!query) {
+    if (
+      hasIssueListRemovalAction &&
+      !this.state.realtimeActive &&
+      actionTakenIds.length > 0
+    ) {
+      const resolvedIds = actionTakenIds
+        .map(id =>
+          this._streamManager
+            .getAllItems()
+            .filter(item => item.id === id && item.status === 'resolved')
+        )
+        .reduce((acc, curr) => acc.concat(curr), [])
+        .map(i => i.id);
+      const ignoredIds = actionTakenIds
+        .map(id =>
+          this._streamManager
+            .getAllItems()
+            .filter(item => item.id === id && item.status === 'ignored')
+        )
+        .reduce((acc, curr) => acc.concat(curr), [])
+        .map(i => i.id);
+      // need to include resolve and ignored statuses because marking as resolved/ignored also
+      // counts as reviewed
+      const reviewedIds = actionTakenIds
+        .map(id =>
+          this._streamManager
+            .getAllItems()
```

Comment:
lets call getAllItems outside the loop

Reply:
since its reused in each one, maybe just have a `const allItems = his._streamManager.getAllItems()` higher up